app/resources/v1/pacbio/request_resource.rb

```diff
@@ -61,17 +89,34 @@ class RequestResource < JSONAPI::Resource
       }
 
       filter :source_identifier, apply: lambda { |records, value, _options|
-        # First we check tubes to see if there are any given the source identifier
-        recs = records.joins(:tube).where(tube: { barcode: value })
-        return recs unless recs.empty?
+        # Initialize an empty result set
+        rec_ids = []
 
-        # If no tubes match the source identifier we check plates
-        # If source identifier specifies a well we need to match samples to well
-        # TODO: The below value[0] means we only take the first value passed in the filter
-        #       If we want to support multiple values in one filter we would need to update this
-        plate, well = value[0].split(':')
-        recs = records.joins(:plate).where(plate: { barcode: plate })
-        well ? recs.joins(:well).where(well: { position: well }) : recs
+        # Iterate over each value in the filter
+        value.each do |val|
+          if val.include?(':')
+            # If the value contains a colon, it's a plate and well identifier
+            plate, well = val.split(':')
+            if plate.present? && well.present?
+              filtered_recs = records.joins(:plate).where(plate: { barcode: plate })
```

Comment:
I think this is a slight regression from before as you could search a plate barcode without a well before. I think thats still useful to have as it should return all requests from that plate.

Reply:
Oh, ok. This is what I mentioned in chat as I was not sure about whether to return all requests in plate if there is no well exists. I will change it.

Reply:
Ah i see. I thought you meant just for malformed ones that had a colon but no well.